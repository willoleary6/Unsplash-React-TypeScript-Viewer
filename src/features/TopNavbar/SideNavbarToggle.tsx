import React, { useRef } from "react";
export const SideNavbarToggle = (): JSX.Element => {
    const body = useRef(document.getElementsByTagName("body")[0]);
    // default keep it closed
    body.current.classList.add("mini-navbar");
    const toggleSideNavbar = () => {
        body.current.classList.toggle("mini-navbar");
    };
    return (
        <a
            id="side-navbar-toggle"
            className="  btn btn-primary"
            onClick={toggleSideNavbar}
            href="#"
        >
            <i className="fa fa-bars fa-lg"></i>
        </a>
    );
};
