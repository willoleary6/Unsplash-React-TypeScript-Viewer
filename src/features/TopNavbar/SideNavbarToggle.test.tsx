import React from "react";
import { SideNavbarToggle } from "./SideNavbarToggle";
import Enzyme, { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
Enzyme.configure({ adapter: new Adapter() });

describe("SideNavbarToggle render testing", () => {
    it("renders panel and takes snapshot to track jsx element changes, will also test for exceptions in element code", () => {
        // using mount here instead of shallow because we want to render the child components too
        const wrapper = shallow(<SideNavbarToggle />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe("SideNavbarToggle functionality testing", () => {
    it("checks that the toggle function is working correctly", () => {
        // using mount here instead of shallow because we want to render the child components too
        const wrapper = shallow(<SideNavbarToggle />);
        wrapper.simulate("click");
        expect(wrapper.hasClass("mini-navbar")).toBe(false);
    });
});
