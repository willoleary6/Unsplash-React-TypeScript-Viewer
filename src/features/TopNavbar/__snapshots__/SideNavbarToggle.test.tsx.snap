// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavbarToggle render testing renders panel and takes snapshot to track jsx element changes, will also test for exceptions in element code 1`] = `
<a
  className="  btn btn-primary"
  href="#"
  id="side-navbar-toggle"
  onClick={[Function]}
>
  <i
    className="fa fa-bars fa-lg"
  />
</a>
`;
